src/libraries/System.Linq.Parallel/src/System/Linq/Parallel/Merging/AsynchronousChannelMergeEnumerator.cs

```diff
@@ -122,7 +122,7 @@ private bool MoveNextSlowPath()
             int firstChannelIndex = _channelIndex;
 
             int currChannelIndex;
-            while ((currChannelIndex = _channelIndex) != _channels.Length)
+            while ((currChannelIndex = _channelIndex) < _channels.Length)
```

Comment:
This is fine, but I highly doubt this will result in any bounds checks being eliminated below.

Reply:
Yeah, I thought some of these were doubtful... at least these changes are more idiomatic - and might help the JIT in future.

Reply:
Bounds check elimination here would depend on at least #11797.